src/simularium/VisGeometry/index.ts

```diff
@@ -879,14 +885,26 @@ class VisGeometry {
                 ((colorNumbers[i] & 0x000000ff) >> 0) / 255.0;
             this.colorsData[i * 4 + 3] = 1.0;
         }
-        this.renderer.updateColors(numColors, this.colorsData);
+    }
 
-        this.visAgents.forEach((agent) => {
-            agent.setColor(
-                this.getColorForTypeId(agent.agentData.type),
-                this.getColorIndexForTypeId(agent.agentData.type)
-            );
-        });
+    public addNewColor(color: number | string): void {
```

Comment:
When users change colors in the ui, how does that result in appending new colors? (I sort of thought the array in here was supposed to be always complete, spanning all possible agent types... maybe that is not true?)

Reply:
I guess that's true, for this implementation I am appending the new colors because I'm not determining from the outset how many of the default colors i'm going to use, but for changing colors we'll probably edit the 4 entries into this array in place. I'll remove this function